exercises/queen-attack/src/example/java/BoardCoordinate.java

```diff
@@ -0,0 +1,39 @@
+public final class BoardCoordinate {
+
+    private final int rank;
+
+    private final int file;
+
+    public BoardCoordinate(final int rank, final int file) throws IllegalArgumentException {
```

Comment:
This is not *at all* a request/demand... just a thought: value objects often come with come with a factory method (usually with a short name).  Then, in places where it is used, they employ a static import.

```java
public static BoardCoordinate at(final int rank, final int file) {
   return new BoardCoordinate(rank, file);
}
```